website/docs/guides/advanced-cluster-new-sharding-schema.html.markdown

```diff
@@ -0,0 +1,344 @@
+        instance_size = "M10"
+        node_count    = 3
+    }
+    provider_name = "AWS"
+    priority      = 7
+    region_name   = "EU_WEST_1"
+    }
+  }
+}
+```
+
+In order to update our configuration to the new schema we will remove the use of `num_shards` and add a new identical `replication_specs` element for each shard. Note that these 2 changes must be done at the same time.
+
+```
+resource "mongodbatlas_advanced_cluster" "test" {
+  project_id = var.project_id
+  name = "GeoShardedCluster"
+  cluster_type   = "GEOSHARDED"
+
+  replication_specs { # first shard for zone n1
+    zone_name  = "zone n1"
+    region_configs {
+    electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+    }
+    provider_name = "AWS"
+    priority      = 7
+    region_name   = "US_EAST_1"
+    }
+  }
+
+  replication_specs { # second shard for zone n1
+    zone_name  = "zone n1"
+    region_configs {
+    electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+    }
+    provider_name = "AWS"
+    priority      = 7
+    region_name   = "US_EAST_1"
+    }
+  }
+
+  replication_specs { # first shard for zone n2
+    zone_name  = "zone n2"
+    region_configs {
+    electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+    }
+    provider_name = "AWS"
+    priority      = 7
+    region_name   = "EU_WEST_1"
+    }
+  }
+
+  replication_specs { # second shard for zone n2
+    zone_name  = "zone n2"
+    region_configs {
+    electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+    }
+    provider_name = "AWS"
+    priority      = 7
+    region_name   = "EU_WEST_1"
+    }
+  }
+}
+```
+
+
+
+This  updated configuration will trigger a terraform update plan, however the underlying cluster will not face any changes after the apply as both configurations represent a geo sharded cluster with 2 zones and 2 shards in each one.
+
+<a id="migration-replicaset"></a>
+## Migrating existing advanced_cluster type REPLICASET
+
+-> **NOTE:**  Please consider the following complementary documentation providing details on transitioning from a replicaset to a sharded cluster: https://www.mongodb.com/docs/atlas/scale-cluster/#convert-a-replica-set-to-a-sharded-cluster.
+
+Considering the following replica set configuration:
+```
+resource "mongodbatlas_advanced_cluster" "test" {
+    project_id   = var.project_id
+    name         = "ReplicaSetTransition"
+    cluster_type = "REPLICASET"
+
+    replication_specs {
+        region_configs {
+            electable_specs {
+                instance_size = "M10"
+                node_count    = 3
+            }
+            provider_name = "AZURE"
+            priority      = 7
+            region_name   = "US_EAST"
+        }
+    }
+}
+```
+
+To transition a replica set to sharded cluster 2 separate updates must be applied. First the cluster type must be adjusted to SHARDED, and apply this change without any other additional changes.
```

Comment:
It seems like there is only one update being made here?

Reply:
Oh, I see. There is another step below. Maybe rephrase to something like:

First, update the cluster type to `SHARDED`, and apply this change to the cluster.